HPCC-14071 Supporting maintaining PackageMaps in multiple files

afishbeck opened the pull request:
Add the ability to add and remove additional files from/to existing
packagemaps.  This will make it easier to support sub collections of
files and queries, and will make it easier for individual ECL
developers to deploy their own packagemaps in shared environments.

Signed-off-by: Anthony Fishbeck anthony.fishbeck@lexisnexis.com

afishbeck committed:
HPCC-14071 Supporting maintaining PackageMaps in multiple files

Add the ability to add and remove additional files from/to existing
packagemaps.  This will make it easier to support sub collections of
files and queries, and will make it easier for individual ECL
developers to deploy their own packagemaps in shared environments.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
@stuartort Please review

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

afishbeck committed:
HPCC-14071 Update based on review

Existing packagemaps will now be converted automatically to the
mulitpart format.

ecl-packagemap-add-part will now create the packagemap if it doesn't
already exist.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

afishbeck commented:
@stuartort updated based on review

Note that:
Existing packagemaps will now be converted automatically to the mulitpart format.
ecl-packagemap-add-part will now create the packagemap if it doesn't already exist.
Merged with changes from the prior dfs update issue.. so please review again.

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

stuartort commented:
these changes look good
